Classes/S3Target.php

```diff
@@ -234,12 +245,16 @@ public function publishCollection(CollectionInterface $collection, callable $cal
             }
         }
 
-        foreach (array_keys($potentiallyObsoleteObjects) as $relativePathAndFilename) {
-            $this->systemLogger->debug(sprintf('Deleted obsolete resource "%s" from bucket "%s"', $relativePathAndFilename, $this->bucketName));
-            $this->s3Client->deleteObject(array(
-                'Bucket' => $this->bucketName,
-                'Key' => $this->keyPrefix . $relativePathAndFilename
-            ));
+        if ($this->unpublishResources !== false) {
+            foreach (array_keys($potentiallyObsoleteObjects) as $relativePathAndFilename) {
+                $this->systemLogger->debug(sprintf('Deleted obsolete resource "%s" from bucket "%s"', $relativePathAndFilename, $this->bucketName));
+                $this->s3Client->deleteObject(array(
+                    'Bucket' => $this->bucketName,
+                    'Key' => $this->keyPrefix . $relativePathAndFilename
+                ));
+            }
+        } else {
```

Comment:
Could turn this around and do just logging and early return in the IF part and then do the unpublishing after outside of the conditionna.

Reply:
will do :) :+1:

Reply:
will do :) :+1: